src/clusterfuzz/_internal/bot/tasks/utasks/analyze_task.py

```diff
@@ -406,29 +420,58 @@ def handle_build_setup_error(output):
         wait_time=testcase_fail_wait)
     return
   data_handler.mark_invalid_uploaded_testcase(
-      output.testcase, output.testcase_upload_metadata, 'Build setup failed')
+      testcase, testcase_upload_metadata, 'Build setup failed')
 
 
 # pylint: disable=no-member
 HANDLED_ERRORS = [
     uworker_msg_pb2.ErrorType.ANALYZE_NO_CRASH,
     uworker_msg_pb2.ErrorType.ANALYZE_BUILD_SETUP,
     uworker_msg_pb2.ErrorType.ANALYZE_NO_REVISIONS_LIST,
-    uworker_msg_pb2.ANALYZE_NO_REVISION_INDEX,
+    uworker_msg_pb2.ErrorType.ANALYZE_NO_REVISION_INDEX,
     uworker_msg_pb2.ErrorType.UNHANDLED,
 ] + setup.HANDLED_ERRORS
 
 # pylint: enable=no-member
 
 
+def _update_testcase(output):
+  """Updates the testcase using the info passed from utask_main."""
+  testcase = data_handler.get_testcase_by_id(output.uworker_input.testcase_id)
```

Comment:
I assume we can trust `uworker_input` here because we always make sure this is the original input we requested right?  @jonathanmetzman

Reply:
Yes. When uworker_output comes from an untrusted source, the trusted receiver derives the input file (which the untrusted party only has read access to) and uses that as input.